src/replica/replica_disk_migrator.cpp

```diff
@@ -310,12 +308,56 @@ bool replica_disk_migrator::migrate_replica_app_info(const replica_disk_migrate_
     return true;
 }
 
-// TODO(jiashuo1)
 // THREAD_POOL_REPLICATION_LONG
-void replica_disk_migrator::close_current_replica() {}
+dsn::task_ptr replica_disk_migrator::close_current_replica(const replica_disk_migrate_request &req)
+{
+    if (_replica->status() != partition_status::type::PS_SECONDARY) {
+        derror_replica("migrate request(origin={}, target={}), err = Invalid partition status({})",
+                       req.origin_disk,
+                       req.target_disk,
+                       enum_to_string(_replica->status()));
+        reset_status();
+        utils::filesystem::remove_path(_target_replica_dir);
+        return nullptr;
+    }
 
-// TODO(jiashuo1)
-// run in replica::close_replica of THREAD_POOL_REPLICATION_LONG
-void replica_disk_migrator::update_replica_dir() {}
+    return _replica->_stub->begin_close_replica(_replica);
+}
+
+// run in replica->close_replica() of THREAD_POOL_REPLICATION_LONG
+void replica_disk_migrator::update_replica_dir()
```

Comment:
The whole process contains: check_args, migrate_data, update_dir, if `update_dir` named `run_migrate`, so what named `migrate_data`? and caller details is:
```
if (replcia_has_moved) {
   update_dir()
}
```
if named `run_migrate`, it will :
```
if (replcia_has_moved) {
   run_migrate()
}
```
it confused that why still `migrate` under the `replica_has_moved`?

so the caller logic `if one replica has been moved, the caller need update the replica dir to new dir` may seem to be right